pkg/cmd/pulumi/state_edit_frontend.go

```diff
@@ -0,0 +1,238 @@
+	if err != nil {
+		return "", err
+	}
+
+	return snapshotText(se.header + "\n" + string(sorted)), nil
+}
+
+func (se *yamlStateFrontend) TextToSnapshot(s snapshotText) (deploy.Snapshot, error) {
+	idk := map[string]interface{}{}
+	err := yaml.Unmarshal([]byte(s), &idk)
+	if err != nil {
+		return deploy.Snapshot{}, err
+	}
+
+	bJSON, err := json.Marshal(idk)
+	if err != nil {
+		return deploy.Snapshot{}, err
+	}
+
+	ctx := context.Background()
+	dep, err := stack.DeserializeUntypedDeployment(ctx, &apitype.UntypedDeployment{
+		Version:    3,
+		Deployment: bJSON,
+	}, stack.DefaultSecretsProvider)
+	if err != nil {
+		return deploy.Snapshot{}, err
+	}
+
+	if dep != nil {
+		return *dep, nil
+	}
+	return deploy.Snapshot{}, errors.New("could not deserialize deployment")
+}
+
+func sortYAML(doc []byte) ([]byte, error) {
+	var node yaml.Node
+	err := yaml.Unmarshal(doc, &node)
+	if err != nil {
+		return []byte{}, err
+	}
+	res := sortYAMLInner(&node)
+	b, err := yaml.Marshal(res)
+	return b, err
+}
+
+func sortYAMLInner(node *yaml.Node) *yaml.Node {
+	if node.Kind == yaml.DocumentNode {
+		for i, n := range node.Content {
+			node.Content[i] = sortYAMLInner(n)
+		}
+	}
+	if node.Kind == yaml.SequenceNode {
+		for i, n := range node.Content {
+			node.Content[i] = sortYAMLInner(n)
+		}
+	}
+	if node.Kind == yaml.MappingNode {
+		for i, n := range node.Content {
+			node.Content[i] = sortYAMLInner(n)
+		}
+		sort.Sort(nodes(node.Content))
+	}
+	return node
+}
+
+type nodes []*yaml.Node
+
+func (i nodes) Less(x, y int) bool {
+	x *= 2
+	y *= 2
+	xKey := i[x].Value
+	yKey := i[y].Value
+	priority := func(key string) int {
+		p, ok := map[string]int{
+			// Resource fields
+			"urn":                     0,
+			"custom":                  1,
+			"delete":                  2,
+			"id":                      3,
+			"type":                    4,
+			"inputs":                  5,
+			"outputs":                 6,
+			"parent":                  7,
+			"protect":                 8,
+			"external":                9,
+			"dependencies":            10,
+			"initErrors":              11,
+			"provider":                12,
+			"propertyDependencies":    13,
+			"pendingReplacement":      14,
+			"additionalSecretOutputs": 15,
+			"aliases":                 16,
+			"customTimeouts":          17,
+			"importID":                18,
+			"retainOnDelete":          19,
+			"deletedWith":             20,
+			"created":                 21,
+			"modified":                22,
+
+			// Update History fields
+			"version":         101,
+			"kind":            102,
+			"result":          103,
+			"message":         104,
+			"resourcecount":   106,
+			"resourcechanges": 105,
+			"starttime":       107,
+			"endtime":         108,
+			"deployment":      109,
+			"environment":     110,
+			"config":          111,
+		}[key]
+		if !ok {
+			return 1000
+		}
+		return p
+	}
+	if priority(xKey) == priority(yKey) {
+		return xKey < yKey
+	}
+	return priority(xKey) < priority(yKey)
```

Comment:
Also, this is lookups for the same key multiple times. We should do them once each.

```
xorder := ...
yorder := ...
if xorder == yorder { return x < y }
return xorder < yorder
```